### PR TITLE
Enable app bypass

### PR DESCRIPTION
**Pull request by bemasc**
This allows a user to select certain apps that should not be
routed through Intra.

This is a workaround for #19, but may also be useful for
advanced users in other situations.

### CODE REVIEW

File: Android/app/src/main/java/app/intra/SettingsFragment.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        // Showing ourselves in the list of apps would be confusing and useless.
+        continue;
+      }
+      entries.add(new Entry(info.loadLabel(pm).toString(), info.packageName));
+    }
+    Collections.sort(entries);
+    labels = new ArrayList<>(entries.size());
+    packageNames = new ArrayList<>(entries.size());
+    for (Entry e : entries) {
+      labels.add(e.label);
```
Comment (cjhenck) [inline Android/app/src/main/java/app/intra/SettingsFragment.java]: Should we use the package name as a backup label?

File: Android/app/src/main/java/app/intra/SettingsFragment.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+      apps.setSummary(R.string.old_android);
+    }
+  }
+
+  /**
+   * Update this SettingsFragment's list of installed apps for the "Excluded apps" setting.  This
+   * method must be called before the fragment is displayed.
+   * @param pm The global PackageManager.
+   */
+  public void updateInstalledApps(PackageManager pm) {
```
Comment (cjhenck) [inline Android/app/src/main/java/app/intra/SettingsFragment.java]: Do you think all installed packages is the most sensible?

I've been debating whether some combo of [LauncherApps](https://developer.android.com/reference/android/content/pm/LauncherApps) and system packages might be a better UX, but I think we wouldn't hit our primary use case (download manager)

Comment (bemasc) [reply Android/app/src/main/java/app/intra/SettingsFragment.java]: Alas, the download manager runs as the app that is using it, so AFAIK there's no way to blacklist it universally.  I did try!

This code uses [`getInstalledApplications`](https://developer.android.com/reference/android/content/pm/PackageManager#getInstalledApplications(int)), not `getInstalledPackages`, so it only collects the "application packages", not "all installed packages".  However, it does seem to be pulling in a lot of packages that aren't normally user-visible.  I added a check to exclude non-launchable packages, which should make the list shorter and more comprehensible.

**Comment by bemasc** [comment]
@cjhenck Please re-review.

**Pull request merged**